code/game/objects/items/stacks/medical.dm

```diff
@@ -85,12 +95,21 @@
 		user.visible_message("<span class='notice'>[user] starts to apply [src] on [user.p_them()]self...</span>", "<span class='notice'>You begin applying [src] on yourself...</span>")
 		if(!do_mob(user, M, self_delay))
 			return
+		//After the do_mob to ensure metabolites have had time to process at least one tick. 
+		if(reagent && (C.reagents.get_reagent_amount(/datum/reagent/metabolite/medicine/styptic_powder) || C.reagents.get_reagent_amount(/datum/reagent/metabolite/medicine/silver_sulfadiazine)))
+			to_chat(user, "<span class='warning'>That stuff really hurt! You can't bring yourself to apply [src] to your wounds again yet, maybe someone else can help put it on for you.</span>")
+			return
 
 	user.visible_message("<span class='green'>[user] applies [src] on [M].</span>", "<span class='green'>You apply [src] on [M].</span>")
-
-	if(affecting.heal_damage(heal_brute, heal_burn) || stop_bleeding)
+	if(reagent)
+		reagents.reaction(M, PATCH, affecting = affecting)
+		M.reagents.add_reagent_list(reagent) //Stack size is reduced by one instead of actually removing reagents from the stack.
```

Comment:
also, please don't use `M.reagents.add_reagent_list(reagent)` directly by taking `reagent` var.
reagent initialisation and reagent transfer to heal should be separated.

Reply:
As I've touched upon in my other responses: Stacks and reagents don't like to get along or play nice.

I'm using the initialized value because the values are effectively acting as flags for how much of what kind of reagent to transfer per use, rather than actually transferring the reagents due to all of the issues and extra snowflake code required for them to do so.